src/OAuthClient/PatronClient.php

```diff
@@ -1,35 +1,77 @@
 <?php
 namespace NYPL\HoldRequestResultConsumer\OAuthClient;
 
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ServerException;
 use NYPL\HoldRequestResultConsumer\Model\DataModel\Patron;
+use NYPL\HoldRequestResultConsumer\Model\Exception\NonRetryableException;
+use NYPL\HoldRequestResultConsumer\Model\Exception\RetryableException;
 use NYPL\Starter\APILogger;
 use NYPL\Starter\Config;
+use NYPL\Starter\Model\Response\ErrorResponse;
 
 class PatronClient extends APIClient
 {
     /**
      * @param string $patronId
-     *
-     * @return Patron | null
+     * @return null|Patron
+     * @throws NonRetryableException
+     * @throws RetryableException
      */
     public static function getPatronById($patronId = '')
     {
         $url = Config::get('API_PATRON_URL') . '/' . $patronId;
 
         APILogger::addDebug('Retrieving patron by id', (array) $url);
 
-        $response = self::get($url);
+        try {
+            $response = self::get($url);
 
-        $response = json_decode((string) $response->getBody(), true);
+            $statusCode = $response->getStatusCode();
 
-        if ($response['statusCode'] !== 200) {
-            APILogger::addError(
-                'Failed',
-                array('Failed to retrieve patron ', $patronId, $response['type'], $response['message'])
+            $response = json_decode((string) $response->getBody(), true);
+
+            APILogger::addDebug(
+                'Retrieved patron by id',
+                $response['data']
             );
-            return null;
-        }
 
-        return new Patron($response['data']);
+            // Check statusCode range
+            if ($statusCode === 200) {
+                return new Patron($response['data']);
+            } else {
+                APILogger::addError(
+                    'Failed',
+                    array('Failed to retrieve patron ', $patronId, $response['type'], $response['message'])
+                );
+                return null;
+            }
+        } catch (ServerException $exception) {
+            throw new RetryableException(
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
+                'Server Error from ' . __FUNCTION__ . $exception->getMessage(),
```

Comment:
Same as above for exception params; second param can be null instead of doubling the message.